### PR TITLE
feature: Support for offset based read & write

### PR DESCRIPTION
**Pull request by jerome89**
Related issue: [#44](https://github.com/graphene-monitoring/graphene/issues/44)
## Offset based metric read & write
Implementation of the following documents:
- [Offset based metric store strategy](https://github.com/graphene-monitoring/graphene/wiki/Concept-of-offset-based-metric-store-strategy-in-Cassandra)
- [Offset based metric read strategy](https://github.com/graphene-monitoring/graphene/wiki/How-to-read-metrics-stored-using-offset-based-strategy-from-Cassandra)

Refactored a lot of classes and restructured project.

### CODE REVIEW

File: graphene-writer/src/main/kotlin/com/graphene/writer/store/GrapheneStoreConfig.kt [hunk 1]
```diff
@@ -1,25 +1,20 @@
...
+//    inputGraphiteCarbonProperty: InputGraphiteCarbonProperty,
+//    handlerProperty: CassandraDataStoreHandlerProperty
+//  ): StoreHandler {
+//
+//    return SimpleDataStoreHandler(
+//      inputGraphiteCarbonProperty,
+//      handlerProperty,
+//      CassandraFactory()
+//    )
+//  }
```
Comment (Dark0096) [inline graphene-writer/src/main/kotlin/com/graphene/writer/store/GrapheneStoreConfig.kt]: How to load data store handler if I set up to OffsetBased or else?

Comment (jerome89) [reply graphene-writer/src/main/kotlin/com/graphene/writer/store/GrapheneStoreConfig.kt]: The store handler is loaded from DataFetchHandlerConfig.kt in the way that is similar to StoreHandlerFactory.kt in Writer. Refactored many classes to have similar concept to Writer for better readability.
This class won't be used anymore, so I will delete the class.

Comment (Dark0096) [reply graphene-writer/src/main/kotlin/com/graphene/writer/store/GrapheneStoreConfig.kt]: I got it :) 
Could you delete this file and resolve conflicts for the merge?

Comment (jerome89) [reply graphene-writer/src/main/kotlin/com/graphene/writer/store/GrapheneStoreConfig.kt]: Done :)

**Pull request merged**